### ISSUE TITLE
Could you please clarify on the email header

### ISSUE THREAD

**Issue by javadbacker** [EvotecIT/Emailimo#1]
How to use the below with outlook email? I have used the below parameters values for sending email using outlook. It is not working for me. Sorry for noobish question.

```powershell
EmailHeader {
        EmailFrom -Address 'mailid@company.com'
        EmailTo -Addresses "another_mailid@company.com"
        EmailServer -Server 'smtp.office365.com' -UserName 'UserName' -Password 'C:\Support\Important\Password-Evotec-Reminder.txt' -PasswordFromFile
        EmailOptions -Priority High -DeliveryNotifications Never
        EmailSubject -Subject 'This is a test email'
    }
```

**Comment by PrzemyslawKlys** [comment]
Hi, this is not your standard Outlook email. You need to know the server, login, and password. It doesn't send thru Outlook. 

Can you clarify exactly what problem are you having so I can help better?

**Comment by javadbacker** [comment]
Thanks for the reply. I agree with your comment on Reddit. HTML composition is a bit difficult (with tables, styling, etc) especially when the data is changing for separate emails. EmailIMO is the way I think.

I ran the script. Initially, it showed error for the password from file. 
the username and password are correct, but I am not sure about the server. 
After google search, I put  smtp.office365.com as the server address. 
I removed the password as secure. Now it runs without error except it doesn't produce any result. 
But I wanted to send email through

**Comment by PrzemyslawKlys** [comment]
Ok, so you want to send an email via Office 365 right?

You need to make sure to put a password in the file thou. Alternatively, if you remove PasswordFromFile switch you can put a password directly. 

```powershell
$UserNotify = 'Przemysław Kłys'
$PasswordExpiryDays = 5

Email -WhatIf -Verbose {
    EmailHeader {
        EmailFrom -Address 'Your@office365Email.com'
        EmailTo -Addresses "YourOtherEmail@com.pl"
        EmailServer -Server 'smtp.office365.com' -UserName 'Your@office365Email.com' -Password 'C:\Support\Important\Password-Euvic-Reminder.txt' -PasswordFromFile
        EmailOptions -Priority High -DeliveryNotifications Never
        EmailSubject -Subject 'This is a test email'
    }
    EmailBody -FontFamily 'Calibri' -Size 15 {
        EmailText -Text "Hello ", $UserNotify, "," -Color None, Blue, None -Verbose -LineBreak
        EmailText -Text "Your password is due to expire in ", $PasswordExpiryDays, "days." -Color None, Green, None
        EmailText -LineBreak
        EmailText -Text 'To change your password: '
        EmailText -Text '- press ', 'CTRL+ALT+DEL', ' -> ', 'Change a password...' -Color None, BlueViolet, None, Red
        EmailText -LineBreak
        EmailTextBox {
            'If you have forgotten your password and need to reset it, you can do this by clicking here. '
            "In case of problems please contact the HelpDesk by visiting [Evotec Website](https://evotec.xyz) or by sending an email to Help Desk."
        }
        EmailText -LineBreak
        EmailText -Text 'Alternatively you can always call ', 'Help Desk', ' at ', '+48 22 00 00 00' `
            -Color None, LightSkyBlue, None, LightSkyBlue -TextDecoration none, underline, none, underline -FontWeight normal, bold, normal, bold
        EmailText -LineBreak
        EmailTextBox {
            'Kind regards,'
            'Evotec IT'
        }
    }
}
```
You can try running this with -Verbose and giving me output to let me know what it says. Remember to fix EmailFrom/EmailTo/EmailServer as per requirements.

**Comment by javadbacker** [comment]
Below is the output I got.

**What if: Performing the operation "Email" on target "To: myid@company.com; CC: ; BCC:".
VERBOSE: Email - Time to send: 0 days, 0 hours, 0 minutes, 0 seconds, 89 milliseconds**

**Comment by javadbacker** [comment]
Code used

_Email -WhatIf -Verbose {
    EmailHeader {
        EmailFrom -Address 'myid@company.com'
        EmailTo -Addresses "myid@company.com"
        EmailServer -Server 'smtp.office365.com' -UserName 'myid@company.com' -Password 'myPasswordString'
        EmailOptions -Priority High -DeliveryNotifications Never
        EmailSubject -Subject 'This is a test email'
    }_

Below is the output I got after directly putting the password



**What if: Performing the operation "Email" on target "To: jckayyath@mcdermott.com; CC: ; BCC:".
VERBOSE: Email - Time to send: 0 days, 0 hours, 0 minutes, 0 seconds, 59 milliseconds**

**Comment by PrzemyslawKlys** [comment]
ok, so now remove WhatIf switch and run again. Make sure to define EmailBody thou. I have never tested without EmailBody.

**Comment by javadbacker** [comment]
I'm always running Email header block and Email body block. 

Below is latest try.

**$UserNotify = 'Przemysław Kłys'
$PasswordExpiryDays = 5

Email -Verbose {
    EmailHeader {
        EmailFrom -Address 'myMailID@mycompany.com'
        EmailTo -Addresses "myMailID@mycompany.com"
        EmailServer -Server 'smtp.office365.com' -UserName 'myMailID@mycompany.com' -Password 'myPasswordString'
        EmailOptions -Priority High -DeliveryNotifications Never
        EmailSubject -Subject 'This is a test email'
    }
    EmailBody -FontFamily 'Calibri' -Size 15 {
        EmailText -Text "Hello ", $UserNotify, "," -Color None, Blue, None -Verbose -LineBreak
        EmailText -Text "Your password is due to expire in ", $PasswordExpiryDays, "days." -Color None, Green, None
        EmailText -LineBreak
        EmailText -Text 'To change your password: '
        EmailText -Text '- press ', 'CTRL+ALT+DEL', ' -> ', 'Change a password...' -Color None, BlueViolet, None, Red
        EmailText -LineBreak
        EmailTextBox {
            'If you have forgotten your password and need to reset it, you can do this by clicking here. '
            "In case of problems please contact the HelpDesk by visiting [Evotec Website](https://evotec.xyz) or by sending an email to Help Desk."
        }
        EmailText -LineBreak
        EmailText -Text 'Alternatively you can always call ', 'Help Desk', ' at ', '+48 22 00 00 00' `
            -Color None, LightSkyBlue, None, LightSkyBlue -TextDecoration none, underline, none, underline -FontWeight normal, bold, normal, bold
        EmailText -LineBreak
        EmailTextBox {
            'Kind regards,'
            'Evotec IT'
        }
    }
}**

_VERBOSE: Performing the operation "Email" on target "To: myMailID@myCompany.com; CC: ; BCC:".
VERBOSE: Email - Time to send: 0 days, 0 hours, 0 minutes, 5 seconds, 638 milliseconds_

**Comment by PrzemyslawKlys** [comment]
Ok, please add:

-Supress $false  it will return Hashtable. 

so:

```powershell
$Test = Email -Supress $False {
...
}
$Test.Status
$Test.Error
$Test.SentTo
```

**Comment by javadbacker** [comment]
Hi. Done as mentioned. Here is the result. Thanks

**$Test.Status**
False

**$Test.SentTo**

**$Test.Error**
Exception calling "Send" with "1" argument(s): "The SMTP server requires a secure connection or the client was not authenticated. The server response was: 5.7.57 SMTP; Client was not authenticated to send anonymous mail during MAIL FROM [SN6PR2101CA0015.namp
rd21.prod.outlook.com]"

**Comment by javadbacker** [comment]
**_I switched on SSL switch and now it works for me._**

**$test = Email -Supress $false {
    EmailHeader {
        EmailFrom -Address 'myMailID@mycompany.com'
        EmailTo -Addresses "myMailID@mycompany.com"
        EmailServer -Server 'smtp.office365.com' -UserName 'myMailID@mycompany.com' -Password 'myPasswordString' -SSL
        EmailOptions -Priority High -DeliveryNotifications Never
        EmailSubject -Subject 'This is a test email'
    }
    EmailBody -FontFamily 'Calibri' -Size 15 {
        EmailText -Text "Hello ", $UserNotify, "," -Color None, Blue, None -Verbose -LineBreak
        EmailText -Text "Your password is due to expire in ", $PasswordExpiryDays, "days." -Color None, Green, None
        EmailText -LineBreak
        EmailText -Text 'To change your password: '
        EmailText -Text '- press ', 'CTRL+ALT+DEL', ' -> ', 'Change a password...' -Color None, BlueViolet, None, Red
        EmailText -LineBreak
        EmailTextBox {
            'If you have forgotten your password and need to reset it, you can do this by clicking here. '
            "In case of problems please contact the HelpDesk by visiting [Evotec Website](https://evotec.xyz) or by sending an email to Help Desk."
        }
        EmailText -LineBreak
        EmailText -Text 'Alternatively you can always call ', 'Help Desk', ' at ', '+48 22 00 00 00' `
            -Color None, LightSkyBlue, None, LightSkyBlue -TextDecoration none, underline, none, underline -FontWeight normal, bold, normal, bold
        EmailText -LineBreak
        EmailTextBox {
            'Kind regards,'
            'Evotec IT'
        }
    }
}**

**Comment by javadbacker** [comment]
Thanks a lot for your time and **Emailimo**

**Comment by PrzemyslawKlys** [comment]
Great! I need to add some debug steps so people can find it on their own :)

**Comment by javadbacker** [comment]
One general question, how much time it took to develop **Emailimo**?

**Comment by PrzemyslawKlys** [comment]
I would say I did in about a week. However I already had PSWriteHTML written, I already had code ready for sending emails, I had concept tested in Dashimo and few other modules so it was more of adjusting the whole concept and HTML to email. I took me 6 months (on and off) to get PSWriteHTML right, to make Statusimo, Dashimo, and Emailimo. But like I said - it's on and off because I get bored easily.
